src/cloud.rs

```diff
@@ -0,0 +1,24 @@
+use core::ptr;
+
+use ll;
+
+pub fn variable(name: &str, variable: &'static i32) -> Result<(), ()> {
+    // null terminated string
+    // the max length of the variable name is 12 characters
+    let mut buffer = [0; 13];
+    buffer[..12][..name.len()].copy_from_slice(name.as_bytes());
+
+    if unsafe {
+        ll::spark_variable(
+            buffer.as_ptr() as *const _ as *const _,
+            variable as *const _ as *const _,
```

Comment:
you can't directly cast from `&T` to `*const U`, you have to do two casts: `&T` -> `*const T` -> `*const U`. This is what this is doing.

And actually I think there's an error here because `c_char` should be `u8` on ARM but I think I defined it as `i8` as if this were for the x86 architecture.